src/hooks/usePendingChangesNotification.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useRef } from 'react';
+import { useRouter } from 'next/navigation';
+import { toast } from 'sonner';
+
+interface PendingChange {
+  details: string;
+}
+
+interface Environment {
+  name: string;
+  pendingChanges: PendingChange[];
+  project?: {
+    name: string;
+  };
+}
+
+interface UsePendingChangesNotificationOptions {
+  // For single environment (EnvironmentPage)
+  environment?: Environment;
+  environmentSlug?: string;
+  
+  // Custom deployment URL (optional)
+  deploymentUrl?: string;
+}
+
+export const usePendingChangesNotification = (options: UsePendingChangesNotificationOptions) => {
+  const router = useRouter();
+  const { environment, environmentSlug, deploymentUrl } = options;
+  const isFirstRender = useRef(true);
+
+  useEffect(() => {
+    const toastId = 'pending-changes';
+    
+    // Handle single environment case
+    if (environment?.pendingChanges && environment.pendingChanges.length > 0) {
+      const defaultDeploymentUrl = `/projects/${environment.project?.name}/${environmentSlug}/deployments`;
+      const finalDeploymentUrl = deploymentUrl || defaultDeploymentUrl;
+      
+      toast.custom(
+        (t) => (
+          <div 
+            className="flex items-center gap-3 p-4 border border-sky-500 rounded-lg shadow-lg min-w-80 max-w-md"
+            style={{ 
+              backgroundColor: '#CFEDFB',
+              color: '#000000'
+            }}
+          >
+            <div className="flex-1">
+              <p className="font-medium text-sm whitespace-nowrap">
+                Changes require deployment to take effect
+              </p>
+            </div>
```

Comment:
As discussed, once the pending changes API has enums for the type, this would be good to get updated to reflect how many of different types are pending.